packages/web3-eth-contract/src/types.ts

```diff
@@ -88,26 +180,308 @@ export type PayableTxOptions = PayableCallOptions;
+
+	/**
+	 * Will send a transaction to the smart contract and execute its method. Note this can alter the smart contract state.
+	 *
+	 * ```ts
+	 * await myContract.methods.myMethod(123).send({from: '0xde0B295669a9FD93d5F28D9Ec85E40f4cb697BAe'});
+	 *
+	 * const receipt = await myContract.methods.myMethod(123).send({from: '0xde0B295669a9FD93d5F28D9Ec85E40f4cb697BAe'});
+	 *
+	 *
+	 * // using the event emitter
+	 * const sendObj = myContract.methods.myMethod(123).send({from: '0xde0B295669a9FD93d5F28D9Ec85E40f4cb697BAe'})
+	 * sendObj.on('transactionHash', function(hash){
+	 *   ...
+	 * });
+	 *
+	 * sendObj.on('confirmation', function(confirmationNumber, receipt){
+	 *   ...
+	 * });
+	 *
+	 * sendObj.on('receipt', function(receipt){
+	 *   // receipt example
+	 *   console.log(receipt);
+	 *   > {
+	 *       "transactionHash": "0x9fc76417374aa880d4449a1f7f31ec597f00b1f6f3dd2d66f4c9c6c445836d8b",
+	 *       "transactionIndex": 0,
+	 *       "blockHash": "0xef95f2f1ed3ca60b048b4bf67cde2195961e0bba6f70bcbea9a2c4e133e34b46",
+	 *       "blockNumber": 3,
+	 *       "contractAddress": "0x11f4d0A3c12e86B4b5F39B213F7E19D048276DAe",
+	 *       "cumulativeGasUsed": 314159,
+	 *       "gasUsed": 30234,
+	 *       "events": {
+	 *           "MyEvent": {
+	 *               returnValues: {
+	 *                   myIndexedParam: 20,
+	 *                   myOtherIndexedParam: '0x123456789...',
+	 *                   myNonIndexParam: 'My String'
+	 *               },
+	 *               raw: {
+	 *                   data: '0x7f9fade1c0d57a7af66ab4ead79fade1c0d57a7af66ab4ead7c2c2eb7b11a91385',
+	 *                   topics: ['0xfd43ade1c09fade1c0d57a7af66ab4ead7c2c2eb7b11a91ffdd57a7af66ab4ead7', '0x7f9fade1c0d57a7af66ab4ead79fade1c0d57a7af66ab4ead7c2c2eb7b11a91385']
+	 *               },
+	 *               event: 'MyEvent',
+	 *               signature: '0xfd43ade1c09fade1c0d57a7af66ab4ead7c2c2eb7b11a91ffdd57a7af66ab4ead7',
+	 *               logIndex: 0,
+	 *               transactionIndex: 0,
+	 *               transactionHash: '0x7f9fade1c0d57a7af66ab4ead79fade1c0d57a7af66ab4ead7c2c2eb7b11a91385',
+	 *               blockHash: '0xfd43ade1c09fade1c0d57a7af66ab4ead7c2c2eb7b11a91ffdd57a7af66ab4ead7',
+	 *               blockNumber: 1234,
+	 *               address: '0xde0B295669a9FD93d5F28D9Ec85E40f4cb697BAe'
+	 *           },
+	 *           "MyOtherEvent": {
+	 *               ...
+	 *           },
+	 *           "MyMultipleEvent":[{...}, {...}] // If there are multiple of the same event, they will be in an array
+	 *       }
+	 *   }
+	 * });
+	 *
+	 * sendObj.on('error', function(error, receipt) { // If the transaction was rejected by the network with a receipt, the second parameter will be the receipt.
+	 *   ...
+	 * });
+	 * ```
+	 *
+	 * @param tx - The options used for sending.
+	 * @returns - Returns a {@link PromiEvent} object resolved with transaction receipt.
+	 */
 	send(
 		tx?: PayableTxOptions,
```

Comment:
```suggestion
		tx?: PayableTxOptions | NonPayableTxOptions,
```

and you can remove line 297 - 299

Reply:
These types are initialised based on the ABI, so one method can either have one send options at same time. That's why can't use `|` here.